sway/tree/view.c

```diff
@@ -450,13 +450,27 @@ static struct sway_container *select_workspace(struct sway_view *view) {
 
 	// Check if there's any `assign` criteria for the view
 	list_t *criterias = criteria_for_view(view,
-			CT_ASSIGN_WORKSPACE | CT_ASSIGN_OUTPUT);
+			CT_ASSIGN_WORKSPACE | CT_ASSIGN_WORKSPACE_NUMBER | CT_ASSIGN_OUTPUT);
 	struct sway_container *ws = NULL;
 	for (int i = 0; i < criterias->length; ++i) {
 		struct criteria *criteria = criterias->items[i];
-		if (criteria->type == CT_ASSIGN_WORKSPACE) {
-			ws = workspace_by_name(criteria->target);
+		if (criteria->type == CT_ASSIGN_OUTPUT) {
+			// CT_ASSIGN_OUTPUT
+			struct sway_container *output = output_by_name(criteria->target);
+			if (output) {
+				ws = seat_get_active_child(seat, output);
+				break;
+			}
+		} else {
+			ws = criteria->type == CT_ASSIGN_WORKSPACE_NUMBER ?
+				workspace_by_number(criteria->target) :
+				workspace_by_name(criteria->target);
+
 			if (!ws) {
+				// We have to create the workspace, but if the container is
+				// sticky and the workspace is going to be created on the same
+				// output, we'll bail out first.
```

Comment:
I copied it over because I'm not sure whether it does apply here.

Consider you have a container that you've assigned as thus:
`for_window [<criteria>] floating enable; sticky enable`
`assign [<criteria>] 2`

and then you open it on workspace 1, then clearly it should open on workspace 1, because it's been stickied. But currently, the code selects workspace 2 to open it on (possibly creating it), before executing the prior commands and realising that it should be stickied, and then ??? before crashing, at least, it does on my machine.

Reply:
At this point the view won't be floating sticky yet because criteria isn't run until near the end of `view_map`. Using your example, I expect the following to happen:

* `select_workspace` returns workspace 2.
* The view is added in the tree as a tiling child of workspace 2 (but no transaction has run yet, so this doesn't appear on screen).
* The criteria runs the `floating` command, which moves the view into the L_FLOATING container of workspace 2.
* The criteria runs the `sticky` command, which should move the view to the currently visible workspace on that output (eg. workspace 1). It doesn't look like the `sticky` command is doing this.
* A transaction is created which makes the final state appear on screen.

Reply:
Okay, I'll remove the comment and see if it can be fixed in the `sticky` command instead.